image: add python3 support for avocado and selenium

jscotka opened the pull request.

martinpitt commented on bots/images/scripts/fedora.setup:
```diff
@@ -115,13 +115,26 @@ dnf $DNF_OPTS -y upgrade
+        chromium-headless \
+        python3-avocado \
+        python3-avocado-plugins-output-html \
+        python3-selenium
```
Do we really need both the python 2 and  3 versions? But I'm okay with cleaning them up in a separate step if that's easier.

martinpitt replied:
Also, we can now drop selenium support from Fedora 28. Your branch also needed rebasing as we recently moved selenium testing to fedora-29.

I did all that in a force-push, and trigger an image refresh now.

cockpituous commented:
image-refresh in progress on bos-tasks-0dnr3.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10544-20181112-080000/

martinpitt commented:
SSL cert got fixed, triggering image.

cockpituous commented:
image-refresh in progress on bos-tasks-3zbvd.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10544-20181112-201414/

cockpituous commented:
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10544-20181112-201414/

martinpitt commented:
Seems one can't just use the fedora-28 repo on fedora-29:
```Problem 1: conflicting requests
  - nothing provides python(abi) = 3.6 needed by python3-avocado-plugins-output-html-62.0-0.fc28.noarch
```
But they do have an f29 repo now, so let's take out the hack.

cockpituous commented:
image-refresh in progress on bos-tasks-bskpf.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10544-20181112-211818/

cockpituous commented:
image-refresh fedora-29 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-29-20181112-220000

martinpitt commented:
Pushed the image.

martinpitt commented:
[container tests fail](https://fedorapeople.org/groups/cockpit/logs/pull-10544-20181112-222424-a1f3340d-container-bastion/log.html) as it seems cockpit/base container is missing from the image:

    repository docker.io/cockpit/base not found: does not exist or no pull access

[avocado test fails](https://fedorapeople.org/groups/cockpit/logs/pull-10544-20181112-222828-a1f3340d-avocado-fedora/log). This looks like this already uses Python 3?

     (1/2) /tmp/avocado_tests/checklogin-basic.py:checklogin_basic.testLogin:  ERROR: 'ascii' codec can't decode byte 0xc3 in position 38: ordinal not in range(128) (0.04 s)

but all the test/avocado/* programs explicitly use python2. This needs to be investigated.

martinpitt commented:
I verified that the avocado test is still being run with Python 2.7. Running the tests with `LC_ALL=C.UTF-8` does not help either.
```
Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:831
Traceback (most recent call last):
  File "/tmp/avocado_tests/checklogin-basic.py", line 48, in testLogin
    c.run_shell_command(setup_cmd, cleanup_cmd)
  File "/tmp/avocado_tests/cockpit.py", line 40, in run_shell_command
    process.run(cmd, shell=True)
  File "/usr/lib/python2.7/site-packages/avocado/utils/process.py", line 1320, in run
    klass = get_sub_process_klass(cmd)
  File "/usr/lib/python2.7/site-packages/avocado/utils/process.py", line 1257, in get_sub_process_klass
    elif should_run_inside_wrapper(cmd):
  File "/usr/lib/python2.7/site-packages/avocado/utils/process.py", line 1231, in should_run_inside_wrapper
    args = cmd_split(cmd)
  File "/usr/lib/python2.7/site-packages/avocado/utils/process.py", line 286, in cmd_split
    data = cmd.encode('utf-8')
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 38: ordinal not in range(128)
```

This keeps happening when I remove python3-{avocado,selenium} packages.

So maybe that's a recent regression from the new python2-avocado in Fedora?

jscotka commented:
> on2. This needs to be inve

@martinpitt  I've met similar issue, could be connected with avocado, that when installed python2/3 it will use avocado3 but should not. I try to find it, and just pytho2-avocado installs avocado binary symlink to ``/usr/bin/avocado -> avocado-2.7``
This is probably caused by new version of avocado what is in upstream.

This issue is caused somehow by ``ssh`` execution. trouble is that ssh is executed via ``env LC_ALL=C ... ssh`` when I've removed LC_ALL  or set to UTF it starts working again. But after few fixes it starts working again also without this setup. I've added this handling to: https://github.com/cockpit-project/cockpit/pull/10486/files#diff-ab6da227e079c0a5e7443b1598980ec1R235

martinpitt commented:
I ported the UTF-8 fixes from PR #10486 and adjusted them a bit. The avocado/selenium tests now work, but the container base issue still remains. Rebuilding the image again for that.

cockpituous commented:
image-refresh in progress on centosci-tasks-kf9hv.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10544-20181113-110505/

cockpituous commented:
image-refresh fedora-29 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-29-20181113-113333

martinpitt commented:
[testPendingClaim failure](https://fedorapeople.org/groups/cockpit/logs/pull-10544-20181113-195656-a860e35d-verify-fedora-29/log.html#7); retrying again to compare
